examples/joystick/Joystick.cpp

```diff
@@ -103,58 +103,43 @@ int main()
     sstr.setf(std::ios::fixed | std::ios::boolalpha);
 
     // Set up our joystick identification sf::Text objects
-    texts["ID"].label.setPosition({5.f, 5.f});
-    texts["ID"].value.setPosition({80.f, 5.f});
-
-    texts["ID"].label.setString("<Not Connected>");
-    texts["ID"].value.setString("");
+    texts.emplace("ID", JoystickObject{{font, "<Not Connected>"}, {font}});
+    texts.at("ID").label.setPosition({5.f, 5.f});
+    texts.at("ID").value.setPosition({80.f, 5.f});
```

Comment:
The return type is pretty ugly. 

```cpp
    auto& id = texts.emplace("ID", JoystickObject{{font, "<Not Connected>"}, {font}}).first->second;
    id.label.setPosition({5.f, 5.f});
    id.value.setPosition({80.f, 5.f});
```
    
I don't expect anyone to understand what `.first->second` means, but I'll use this pattern instead if you prefer it.

Reply:
I wanted to suggest structured bindings, but then it came to my mind that there's no way to ignore a part of it.
Why is every feature always half-done :disappointed_relieved:

It's probably still better to not throw that information away instead of doing the lookup twice, not sure how others see this...

Reply:
```cpp
    auto& [id_key, id] = *texts.emplace("ID", JoystickObject{{font, "<Not Connected>"}, {font}}).first;
    id.label.setPosition({5.f, 5.f});
    id.value.setPosition({80.f, 5.f});
```
Here I'm destructuring the pair that was just emplaced. It's a bit awkward to pick variable names that won't conflict with the `.at("Threshold")` snippet a few lines down which will need the same treatment. I can't use a generic variable name like `key` without shadowing warnings breaking the build.

Because this is example code I don't feel strongly and will change it to one of these alternative options if you prefer. I now see we're already using the `.first->second` pattern elsewhere in this file so I'm less reluctant to add more of it.

Reply:
+1 for destructuring, but I would do it like this for readability:

```cpp
auto [it, success] = texts.emplace("ID", JoystickObject{{font, "<Not Connected>"}, {font}});
auto& [label, value] = *it;
label.setPosition({5.f, 5.f}); 
value.setPosition({80.f, 5.f});
```

Or just this:

```cpp
auto [it, success] = texts.emplace("ID", JoystickObject{{font, "<Not Connected>"}, {font}});
it->label.setPosition({5.f, 5.f}); 
it->value.setPosition({80.f, 5.f});
```